### PR TITLE
Hoe upgrades

### PR DESCRIPTION
**Pull request by jeremyckahn**
### What this PR does

It implements Hoe upgrades (#222)! Upgraded Hoes can potentially reclaim the seeds that were planted in a given plot. Each upgrade increases the chance of seed retrieval, all the way up to 100% for the Gold Hoe.

This PR also fixes and improves some tangentially-related parts of the code, which will be called out inline.

### How this change can be validated

Preview URL: https://farmhand-git-jeremyckahn-222hoe-upgrades-jeremyckahn.vercel.app/

Here's the minimal state needed to upgrade the Hoe:

```js
farmhand.setState({
  money: 10000000,
  purchasedSmelter: 1,
  inventoryLimit: -1,
  inventory: [
    {
      id: 'coal',
      quantity: 1000,
    },
    {
      id: 'gold-ingot',
      quantity: 1000,
    },
    {
      id: 'bronze-ingot',
      quantity: 1000,
    },
    {
      id: 'iron-ingot',
      quantity: 1000,
    },
    {
      id: 'silver-ingot',
      quantity: 1000,
    },
  ],
  learnedRecipes: {
    'bronze-ingot': true,
    'iron-ingot': true,
    'silver-ingot': true,
    'gold-ingot': true,
  },
})
```

Basically, you just need all of the ingots learned and in stock. From there you can easily upgrade the Hoe in the Forge. From there, plant some seeds, use the Hoe on them, and see them return to your inventory appropriate for the level of Hoe you have.

### Questions or concerns about this change

Any thoughts on the recipe ingredients for the different Hoe levels? I mostly copy/pasted from one to the next, but I'm open-minded to more ingredient quantity variation.

### Additional information

Upgradable Hoe

Updated plot highlight colors when the Hoe is selected

### CODE REVIEW

File: src/components/Plot/Plot.sass [hunk 1]
```diff
@@ -50,7 +50,7 @@ $not-allowed-color: rgba(255, 0, 0, 0.5)
     cursor: pointer
 
   .cleanup-mode &.is-ripe
-    background-color: var(--color-red-danger)
+    background-color: var(--color-green-ok)
```
Comment (jeremyckahn) [inline src/components/Plot/Plot.sass:53]: Using the Hoe on a mature crop now returns to the inventory.

File: src/components/Toolbelt/Toolbelt.js [hunk 2]
```diff
@@ -20,17 +20,15 @@ import './Toolbelt.sass'
...
-  let tools = []
+  const tools = []
 
   for (let tool of Object.values(toolsData)) {
     if (toolLevels[tool.type] !== toolLevel.UNAVAILABLE) {
       tools.push(tool)
     }
   }
 
-  tools.sort(t => t.order)
```
Comment (jeremyckahn) [inline src/components/Toolbelt/Toolbelt.js:31]: This was actually broken previously. I'm not sure when; I think I introduced this bug. The broken sorting was breaking keyboard combos, so this will be nice to ship a fix for!

File: src/event-handlers.js [hunk 3]
```diff
@@ -359,8 +359,10 @@ export default {
         }
 
         this.setState({
-          ...this.createInitialState(),
-          ...transformStateDataForImport(state),
+          ...transformStateDataForImport({
+            ...this.createInitialState(),
+            ...state,
+          }),
```
Comment (jeremyckahn) [inline src/event-handlers.js:365]: This fixes broken data imports for some variations of old Farmhand game data.

**Review by jeremyckahn — COMMENTED** [review]
I left some notes inline to hopefully help give some context around some of the changes!

**Review by lstebner — COMMENTED** [review]
this is looking good to me! I have one small question, but that's it. I think the recipe ingredients are fine, though I'm also open to lowering them slightly since they're the same as the scythe, but the shovel takes more. I don't think I would go below 8 though? nice work!

File: src/reducers.js [hunk 4]
```diff
@@ -1797,6 +1787,14 @@ export const minePlot = (state, x, y) => {
   }
 }
 
+const hoeLevelToSeedReclaimRate = {
```
[the remark this answers is not in the document]
Comment (jeremyckahn) [reply src/reducers.js:1790]: I think so; it's meant to to be an unchanging lookup map. Do you have something else in mind for what this should be?

**Comment by jeremyckahn** [comment]
> this is looking good to me! I have one small question, but that's it. I think the recipe ingredients are fine, though I'm also open to lowering them slightly since they're the same as the scythe, but the shovel takes more. I don't think I would go below 8 though? nice work!

That makes sense. I've lowered to 8 just to have some differentiation among the levels in https://github.com/jeremyckahn/farmhand/commit/297b1dbb6099ba3914117b688518fb8af73e1aaf!

File: src/reducers.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (lstebner) [reply src/reducers.js:1790]: sorry, what I meant was- does it belong in the file with the other constants? not the 'const' definition, my bad!

[the remark this answers is not in the document]
Comment (jeremyckahn) [reply src/reducers.js]: Ah yeah, I went back and forth on that. I'm fine with moving it. Done in https://github.com/jeremyckahn/farmhand/commit/5a4922e6c76b69d2f9113d372d80e1567fddddab!

**Review by lstebner — APPROVED** [review]
🎉
